streams/src/main/java/org/apache/kafka/streams/AutoOffsetReset.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams;
+
+import java.time.Duration;
+import java.util.Optional;
+
+/**
+ * Sets the {@code auto.offset.reset} configuration when
+ * {@link #addSource(AutoOffsetReset, String, String...) adding a source processor} or when creating {@link KStream}
+ * or {@link KTable} via {@link StreamsBuilder}.
+ */
+public class AutoOffsetReset {
+
+    private enum OffsetResetType {
+        LATEST,
+        EARLIEST,
+        BY_DURATION
+    }
+
+    private final OffsetResetType type;
+    private final Optional<Long> duration;
+
+    private AutoOffsetReset(OffsetResetType type, Optional<Long> duration) {
+        this.type = type;
+        this.duration = duration;
+    }
+
+    /**
+     * Creates an AutoOffsetReset instance representing "latest".
+     * 
+     * @return an AutoOffsetReset instance for the "latest" offset.
+     */
+    public static AutoOffsetReset latest() {
+        return new AutoOffsetReset(OffsetResetType.LATEST, Optional.empty());
+    }
+
+    /**
+     * Creates an AutoOffsetReset instance representing "earliest".
+     * 
+     * @return an AutoOffsetReset instance for the "earliest" offset.
+     */
+    public static AutoOffsetReset earliest() {
+        return new AutoOffsetReset(OffsetResetType.EARLIEST, Optional.empty());
+    }
+
+    /**
+     * Creates an AutoOffsetReset instance with a custom duration.
+     * 
+     * @param duration the duration to use for the offset reset; must be non-negative.
+     * @return an AutoOffsetReset instance with the specified duration.
+     * @throws IllegalArgumentException if the duration is negative.
+     */
+    public static AutoOffsetReset byDuration(final Duration duration) {
+        if (duration.isNegative()) {
+            throw new IllegalArgumentException("Duration cannot be negative");
+        }
+        return new AutoOffsetReset(OffsetResetType.BY_DURATION, Optional.of(duration.toMillis()));
+    }
+
+    /**
+     * Provides a human-readable description of the offset reset type and duration.
+     * 
+     * @return a string describing the offset reset configuration.
+     */
+    public String describe() {
```

Comment:
I don't think we need anything like this -- also, if we would want to add it, it would need to be part of the KIP.

I would just remove it and not alter the KIP. If there is demand in the future, we can still do a follow up KIP an add it. But my gut feeling is, we won't need it, and/or could just add an overwrite for `toString()`.

Reply:
Sounds good, will remove it @mjsax. Next time I will present it as part of discussion or ask in ongoing conversation before adding something ad-hoc like this